core/src/main/java/org/elasticsearch/node/internal/InternalSettingsPreparer.java

```diff
@@ -54,6 +54,8 @@
     private static final String[] ALLOWED_SUFFIXES = {".yml", ".yaml", ".json", ".properties"};
     static final String[] PROPERTY_PREFIXES = {"es.", "elasticsearch."};
     static final String[] PROPERTY_DEFAULTS_PREFIXES = {"es.default.", "elasticsearch.default."};
+    // this is used to filter out things like `es.tests.portfile` which should only be accessed via system properties directly
+    static final String[] PROPERTY_DEFAULTS_AND_TEST_PREFIXES = {"es.default.", "elasticsearch.default.", "es.tests",  "elasticsearch.tests"};
```

Comment:
I agree with you... I think the better fix is to make it not start with `es.` and just use `test.portfile` then it will not end up here?